sfputil/debug.py

```diff
@@ -0,0 +1,149 @@
+import sys
+import click
+from sonic_py_common import multi_asic
+import utilities_common.cli as clicommon
+from utilities_common import platform_sfputil_helper
+from utilities_common.platform_sfputil_helper import (
+    is_rj45_port,
+    is_sfp_present,
+    get_subport,
+    get_sfp_object,
+    get_subport_lane_mask
+)
+from swsscommon.swsscommon import SonicV2Connector, ConfigDBConnector
+
+EXIT_FAIL = -1
+EXIT_SUCCESS = 0
+ERROR_PERMISSIONS = 1
+ERROR_CHASSIS_LOAD = 2
+ERROR_SFPUTILHELPER_LOAD = 3
+ERROR_PORT_CONFIG_LOAD = 4
+ERROR_NOT_IMPLEMENTED = 5
+ERROR_INVALID_PORT = 6
+
+
+@click.group(cls=clicommon.AliasedGroup)
+def debug():
+    """
+    Group for debugging and diagnostic control commands.
+
+    This command group loads platform-specific utilities and prepares them for use in diagnostic commands.
+    """
+    platform_sfputil_helper.load_platform_sfputil()
+    platform_sfputil_helper.load_chassis()
+    platform_sfputil_helper.platform_sfputil_read_porttab_mappings()
+
+
+@debug.command()
+@click.argument('port_name', required=True)
+@click.argument(
+    'loopback_mode',
+    required=True,
+    type=click.Choice(["host-side-input", "host-side-output", "media-side-input", "media-side-output"])
+)
+@click.argument('enable', required=True, type=click.Choice(["enable", "disable"]))
+def loopback(port_name, loopback_mode, enable):
+    """
+    Set module diagnostic loopback mode.
+    """
+    sfp = get_sfp_object(port_name)
+
+    if is_rj45_port(port_name):
+        click.echo(f"{port_name}: This functionality is not applicable for RJ45 port")
+        sys.exit(EXIT_FAIL)
+
+    if not is_sfp_present(port_name):
+        click.echo(f"{port_name}: SFP EEPROM not detected")
+        sys.exit(EXIT_FAIL)
+
+    try:
+        api = sfp.get_xcvr_api()
+    except NotImplementedError:
+        click.echo(f"{port_name}: This functionality is not implemented")
+        sys.exit(ERROR_NOT_IMPLEMENTED)
+
+    namespace = multi_asic.get_namespace_for_port(port_name)
+    config_db = ConfigDBConnector(use_unix_socket_path=True, namespace=namespace)
+    config_db.connect()
+
+    try:
+        subport = int(config_db.get(config_db.CONFIG_DB, f'PORT|{port_name}', 'subport'))
+    except TypeError:
+        click.echo(f"{port_name}: subport is not present in CONFIG_DB")
+        sys.exit(EXIT_FAIL)
+
+    subport = max(subport, 1)
+
+    state_db = SonicV2Connector(use_unix_socket_path=False, namespace=namespace)
+    state_db.connect(state_db.STATE_DB)
+
+    try:
+        host_lane_count = int(state_db.get(state_db.STATE_DB, f'TRANSCEIVER_INFO|{port_name}', 'host_lane_count'))
```

Comment:
@vdahiya12 Can we instead use `get_value_from_db_by_field?

Reply:
fixed